### PR TITLE
test(compat/mergeWith): add missing test cases for edge cases and improve coverage to 100%

### PR DESCRIPTION
**Pull request by wo-o29**
## Summary  
  
This PR improves test coverage for `compat/mergeWith` function in the compat module from 85.41% to 100% by adding comprehensive test cases for previously uncovered edge cases.  
  
## Changes  
  
- Added test for prototype pollution prevention (`__proto__` handling)  
- Added test for circular reference handling in source objects  
- Added test for Arguments objects in both source and target  
- Added test for Buffer object cloning  
- Added test for TypedArray cloning  
- Added test for handling undefined values in both target and source  

## Impacts
| Before | After |
|:------:|:-----:|
|  스크린샷 2025-11-15 오후 1 40 42|  스크린샷 2025-11-15 오후 1 39 51|

### CODE REVIEW

File: src/compat/object/mergeWith.spec.ts [hunk 1]
```diff
@@ -159,4 +159,73 @@ describe('mergeWith', () => {
...
+    expect(result.circular).toBeDefined();
+    expect(result.circular.a).toBe(1);
+  });
+
+  it('should handle Arguments objects in both source and target', () => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    function getArgs(..._arg: unknown[]) {
+      // eslint-disable-next-line prefer-rest-params
+      return arguments;
+    }
```
Comment (wo-o29) [inline src/compat/object/mergeWith.spec.ts:190]: I've managed to bypass the lint rules for unused variables and arguments objects, is there a better way?

Comment (raon0211) [reply src/compat/object/mergeWith.spec.ts:190]: I think this is the best way to handle this so far.

**Review by raon0211 — APPROVED** [review]
Thanks so much!

File: src/compat/object/mergeWith.spec.ts (hunk 1 above)
Comment (wo-o29) [reply src/compat/object/mergeWith.spec.ts:190]: Thank you for your good opinon!
